ftw/publisher/sender/taskqueue/queue.py

```diff
@@ -46,7 +46,12 @@ def __call__(self):
         obj = api.portal.get().unrestrictedTraverse(path, None)
 
         require_token = self.request.form['token']
-        current_token = IAnnotations(obj).get(TOKEN_ANNOTATION_KEY, None)
+
+        if obj is None:
+            current_token = None
```

Comment:
It's kind of a side effect that this will result in a retry, but I see that there is no nice way to do that without rewriting / extracting parts here.

For making it clearer what's happening here for the future us I suggest to add a comment:
```python

        if obj is None:
            # The object does not yet exist under this path.
            # The reason is most likely that we are processing a move job
            # which was created in another process, which has not yet finished
            # committing.
            # We want to retry so that we can operate on the transaction where
            # the object is already moved to the target.
            # In order to trigger a retry, we set the ``current_token`` to
            # ``None``.
            current_token = None
```